python/ray/tune/tests/test_result_grid.py

```diff
@@ -224,6 +224,62 @@ def f(config):
     assert not any(key in representation for key in AUTO_RESULT_KEYS)
 
 
+def test_result_grid_repr(ray_start_2_cpus, tmpdir):
+    from ray.air.result import Result
+
+    results = list()
+    results.append(
+        Result(
+            metrics={"loss": 1.0},
+            checkpoint=Checkpoint(data_dict={"weight": 1.0}),
+            log_dir=Path("./log_1"),
+            error=None,
+            metrics_dataframe=None,
+            best_checkpoints=None,
+        )
+    )
+    results.append(
+        Result(
+            metrics={"loss": 2.0},
+            checkpoint=Checkpoint(data_dict={"weight": 2.0}),
+            log_dir=Path("./log_2"),
+            error=RuntimeError(),
+            metrics_dataframe=None,
+            best_checkpoints=None,
+        )
+    )
+
+    result_grid = ResultGrid(experiment_analysis=None)
```

Comment:
Ah, I see why we had the check if `experiment_analysis` is `None` now. We can just create a mock analysis object that contains an empty list of trials. This way, we don't need to change the experiment analysis to be an optional parameter.

```python
class MockExperimentAnalysis:
    trials = []
```